### ISSUE TITLE
Dim only Chromium window

### ISSUE THREAD

**Issue by rncar** [yshui/compton#925]
Hi, picom works like a charm in my crappy integrated gpu at 4K!!

But I cant find a way to do this. The browser are killing my eyes and I want to low his brightness.
I cant use the maximun brightness option because the glx backend is too heavy.

Would be great to dim only the Chromium window or only a especific desktop, 
I always open the browser in the desktop 9.

Any way to do it ?

Thanks a lot.

**Comment by yshui** [comment]
hmm, we don't have a `dim-rule` yet. could be a good idea.

in the mean time you can write a shader and use it with `window-shader-fg-rule`

**Comment by rncar** [comment]
The `dim-rule` would be great. 

I think I cant use `window-shader-fg-rule` because I am using the `xrender` backend.
(less heavy than `glx` in my gpu)

**Comment by yshui** [comment]
@rncar that's surprising. what's your hardware? i won't expect glx to be more heavy unless you enabled fancy effects.

`xrender` uses the same hardware acceleration as opengl nowadays.

**Comment by rncar** [comment]
It's the Intel Core i5-12600 integrated gpu (Intel Alderlake_s Gen12).
`OpenGL renderer string: Mesa Intel(R) UHD Graphics 770 (ADL-S GT1)`

**Comment by yshui** [comment]
That should be enough to run picom

**Comment by rncar** [comment]
Yes, picom runs great in both, but the difference is in opengl/vulkan.
For example, ~55% gpu usage in `glxgears` for `xrender`, ~85% in `glx`.

`xrender` backend feels more snappy for this kind of applications for this gpu/driver combination.
I am a graphics programmer, I am running opengl apps all day long ; )

**Comment by rncar** [comment]
In any case, I think to add a `dim-rule` could be a good idea.
And thanks for your prompt responses.

**Comment by rncar** [comment]
Any way to change brightness/dim dynamically using dbus ?

**Comment by absolutelynothelix** [comment]
@rncar, looks like it's possible to change some options dynamically via dbus, but i can't find a suitable option for you.

**Comment by rncar** [comment]
Yes, it seems so. I came to the same conclusion.
